tests/conftest.py

```diff
@@ -164,12 +164,16 @@ def add_file_change_commit(
         if parent_folder:
             os.makedirs(parent_folder, exist_ok=True)
 
-        with open(
-            os.path.join(self.working_dir, file_name), "w", encoding=encoding
-        ) as output_file:
-            output_file.write(contents)
+        file_path = os.path.join(self.working_dir, file_name)
+
+        if contents is None:
+            os.unlink(file_path)
+            self.index.remove(file_name)
+        else:
+            with open(file_path, "w", encoding=encoding) as output_file:
+                output_file.write(contents)
+            self.index.add([file_name])
```

Comment:
Writing this test turned out trickier than i expected, but look at this.

We need to be able to commit deleted files, so i added this if/else block. But, maybe just for clarity i should rewrite this as separate "add_file_delete_commit"?

Reply:
yeah, I think `add_file_delete_commit` makes a lot of sense!